woocommerce/payment-gateway/class-sv-wc-payment-gateway-my-payment-methods.php

```diff
@@ -245,16 +233,64 @@ public function render_js() {
 			 */
 			$args = apply_filters( 'wc_payment_gateway_' . $this->get_plugin()->get_id() . '_payment_methods_js_args', $args, $this );
 
-			wc_enqueue_js( sprintf(
-				'window.wc_%1$s_payment_methods_handler = new %2$s( %3$s );',
-				esc_js( $this->get_plugin()->get_id() ),
-				esc_js( parent::get_js_handler_class_name() ),
-				json_encode( $args )
-			) );
+			ob_start();
+
+			$window_object = 'wc_' . esc_js( $this->get_plugin()->get_id() ) . '_payment_methods_handler';
+			$load_function = 'load_' . esc_js( $this->get_plugin()->get_id() ) . '_payment_methods_handler';
+			$loaded_event  = strtolower( $this->get_js_handler_class_name() ) . '_loaded';
```

Comment:
That's a good idea. Updated in 48d12a2b.

An idea for another story: the snippet is also almost identical in the three cases, except for what happens after the handler is instantiated (triggering `update_checkout` or calling `init()`). I think it could be useful to have a helper method on the parent class that renders the `try/catch` block and accepts extra code to add inside the load function. The class name and other variables could be parameters of that helper method.